core/types/src/error.rs

```diff
@@ -1,9 +1,375 @@
 // Copyright (c) 2022 The MobileCoin Foundation
 
+//! SGX Error types
+
+use core::convert::From;
+use mc_sgx_core_sys_types::sgx_status_t;
+use serde::{Deserialize, Serialize};
+
 /// Errors seen when converting to, or from, rust for the SGX types
 #[derive(Debug)]
-pub enum Error {
+pub enum FfiError {
     /// Enum out of range
     /// Happens when a value that is not represented by the known C enum values.
     UnknownEnumValue(i64),
 }
+
+/// A enumeration of SGX errors.
+///
+/// Those listed here are the ones which are identified in the `sgx_status_t`
+/// enum, in order of the actual value. Note that values are grouped
+/// (numerically) into the following general sections:
+///
+///  1. `0x0000-0x0fff`: Generic errors.
+///  2. `0x1000-0x1fff`: Fatal runtime errors.
+///  3. `0x2000-0x2fff`: Enclave creation errors.
+///  4. `0x3000-0x3fff`: Local attestation/report verification errors.
+///  5. `0x4000-0x4fff`: Errors when communicating with the Architectural
+///                      Enclave Service Manager (AESM).  
+///  6. `0x5000-0x5fff`: Errors internal to AESM.  
+///  7. `0x6000-0x6fff`: Errors with the encrypted enclave loader.
+///  8. `0x7000-0x7fff`: Errors with the "SGX Encrypted FS" utility.
+///  9. `0x8000-0x8fff`: Attestation key errors.
+/// 10. `0xf000-0xffff`: Internal (to SGX) errors.
+#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize, Deserialize)]
```

Comment:
I *think* you can do `#[cfg_attr(feature = "serde", derive(Deserialize, Serialize)]` as a second line here and it will do the appropriate feature gating.

Reply:
done